### ISSUE TITLE
Convert Json to Csv vice versa

### ISSUE THREAD

**Issue by bezaleluriel** [Cinchoo/ChoETL#115]
Hi,
I tried to convert json file to csv file vice versa using ChoETL.
the conversion from json to csv was perfect!
but the conversion from csv to json, was not as I expected. (I wanted it to be just like the json original file)

**my original json:**

  {
    "str1": "aaa",
    "num1": 1,
    "boolean1": true,
    "object1": {
        "str2": "bbb",
        "num2": 2,
        "boolean2": false
    },
    "list1": [
    	"ccc",
        "ddd",
        "eee"
    ],
    "list2": [
        1,
        2,
        3
    ],
    "list3": [
    	true,
        false,
        true
    ]
}

**function of json to csv:**

StringBuilder csv = new StringBuilder();
string json = File.ReadAllText("jsonInputExample.json");
using (var p = new ChoJSONReader(new StringReader(json))
{
    using (var w = new ChoCSVWriter(new StringWriter(csv))
    .WithFirstLineHeader()
    .NestedColumnSeparator('.')
    )
    {
        w.Write(p);
    }
}
File.WriteAllText("csvOutputExample.csv", csv.toString());  

**output of csv:** (exactly what I meant)

str1,num1,boolean1,object1.str2,object1.num2,object1.boolean2,list1.0,list1.1,list1.2,list2.0,list2.1,list2.2,list3.0,list3.1,list3.2
aaa,1,True,bbb,2,False,ccc,ddd,eee,1,2,3,True,False,True 

**function of csv to json:**

StringBuilder json = new StringBuilder();
//the csv output  from the first function
string csv= File.ReadAllText("csvOutputExample.csv");
using (var p = new ChoCSVReader(new StringReader(csv))
         .NestedColumnSeparator('.')
         . WithFirstLineHeader()
     )
{
    using (var w = new ChoJSONWriter(new StringWriter(json))
         .SupportMultipleContent(true)
    )
    {
        w.Write(p);
    }
}
File.WriteAllText("jsonOutputExample.csv", json.toString());  

**output of json:**

{
 {
    "str1": "aaa",
    "num1": "1",
    "boolean1": "true",
    "object1": {
        "str2": "bbb",
        "num2": "2",
        "boolean2": "false"
    },
    "list1": {
    	"0" : "ccc",
        "1" : "ddd",
        "2" : "eee"
    },
    "list2": {
       "0" : "1",
       "1" : "2",
       "2" :  "3"
    },
    "list3": {
    	"0" :"true",
        "1" : "false",
        "2" :"true"
    }
 }
}

**The problems:**

1. Added one more bracelets on the outside 
2. All values look like strings (instead of int or boolean..)
3. all the array objects look like Object instead of array (wrong type of bracelets, key of "0", "1", "2"..)

How can I fix the code so that the output file looks exactly like the original  json file??

Thank you,
Uriel

**Comment by Cinchoo** [comment]
Ran test, found couple of issues. Applied fixes and pushed new package v1.2.1.3

try it and let me know.

```
string json = @"
{
  ""str1"": ""aaa"",
  ""num1"": 1,
  ""boolean1"": true,
  ""object1"": {
    ""str2"": ""bbb"",
    ""num2"": 2,
    ""boolean2"": false
  },
  ""list1"": [
    ""ccc"",
    ""ddd"",
    ""eee""
  ],
  ""list2"": [
    1,
    2,
    3
  ],
  ""list3"": [
    true,
    false,
    true
  ]
}";

StringBuilder csv = new StringBuilder();
using (var p = ChoJSONReader.LoadText(json)
    .UseJsonSerialization()
    )
{
    using (var w = new ChoCSVWriter(csv)
    .WithFirstLineHeader()
    .NestedColumnSeparator('.')
    .ArrayIndexSeparator('_')
    )
    {
        w.Write(p);
    }
}
Console.WriteLine(csv.ToString());


StringBuilder json1 = new StringBuilder();
using (var r = ChoCSVReader.LoadText(csv.ToString())
    .WithFirstLineHeader()
    .NestedColumnSeparator('.')
    .ArrayIndexSeparator('_')
    .WithMaxScanRows(2)
    )
{
    using (var w = new ChoJSONWriter(json1)
        )
    {
        w.Write(r);
    }
}
Console.WriteLine(json1.ToString());
```

**Comment by bezaleluriel** [comment]
Thank you!
Works well for me.
